src/main/java/net/tridentsdk/chat/ChatComponent.java

```diff
@@ -0,0 +1,482 @@
+/*
+ * Trident - A Multithreaded Server Alternative
+ * Copyright 2016 The TridentSDK Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.tridentsdk.chat;
+
+import com.google.common.collect.Lists;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import lombok.*;
+import lombok.experimental.Accessors;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Represents a component in a Minecraft chat format.
+ */
+@Accessors(chain = true)
+@NoArgsConstructor
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class ChatComponent {
+
+    @Getter @Setter
+    private String text, translate, selector, insertion, scoreUsername, scoreObjective;
+
+    @Getter @Setter
+    private ChatColor color;
+
+    @Getter @Setter
+    private ClickEvent clickEvent;
+
+    @Getter @Setter
+    private HoverEvent hoverEvent;
+
+    private List<JsonElement> with = Lists.newCopyOnWriteArrayList();
+    private List<ChatComponent> extra = Lists.newCopyOnWriteArrayList();
+    private AtomicBoolean bold, italic, underlined, strikethrough, obfuscated;
+
+    private ReadWriteLock $lock = new ReentrantReadWriteLock();
+
+    /**
+     * Gets all elements attached to the 'with' array.
+     *
+     * The 'with' array is used in conjunction with the 'translate' component.
+     *
+     * @return The with elements.
+     */
+    public List<JsonElement> getWith() {
+        return Collections.unmodifiableList(with);
+    }
+
+    /**
+     * Adds an element to the 'with' array.
+     *
+     * @param element The JSON element.
+     *
+     * @return This object.
+     */
+    public ChatComponent addWith(JsonElement element) {
+        $lock.writeLock().lock();
```

Comment:
THREAD-SAFETY
You need to make up your mind, use locks OR the thread-safe collection